### PR TITLE
Return correct type indices for primitive types referred to by their CLR type names

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

Return correct type indices for primitive types referred to by their CLR type names - this permits a node like `List.RemoveIfNot` to work with not just the DS types passed as input, but also the C# type. For e.g. it now works for both the DS type `int` but also the equivalent C# type, `System.Int64` type.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/Engine/ProtoCore/Lang/Type.cs [hunk 1]
```diff
@@ -121,6 +121,16 @@ public class TypeSystem
         public Dictionary<ProtoCore.DSASM.AddressType, int> addressTypeClassMap { get; set; }
         private static Dictionary<PrimitiveType, string> primitiveTypeNames;
 
+        private static readonly Dictionary<string, string> clrToDSTypeMap = new Dictionary<string, string>
+        {
+            { $"{nameof(System)}.{nameof(String)}", "string" },
+            { "long", "int" },
```
Comment (mjkkirschner) [inline src/Engine/ProtoCore/Lang/Type.cs:127]: where does the `long` type name come from given that `int64` is the actual .net type name?

Comment (mjkkirschner) [reply src/Engine/ProtoCore/Lang/Type.cs:127]: considering that this dictionary is only created once - maybe it's not so bad to do:
`typeof(Int64).FullName`

**Review by mjkkirschner — APPROVED** [review]

File: src/Engine/ProtoCore/Lang/Type.cs [hunk 2]
```diff
@@ -314,7 +324,13 @@ public string GetType(int UID)
         public int GetType(string ident)
         {
             Validity.Assert(null != classTable);
-            return classTable.IndexOf(ident);
+            var index = classTable.IndexOf(ident);
+            if (index != Constants.kInvalidIndex) return index;
+
+            if(clrToDSTypeMap.TryGetValue(ident, out string dsType))
+                return classTable.IndexOf(dsType);
```
Comment (mjkkirschner) [inline src/Engine/ProtoCore/Lang/Type.cs:331]: while I think this PR looks good - I am curious if you think there could be performance implications here, this `IndexOf` call doesn't seem to be straight forward, it seems to do partial matching? So in the worst case we'll now do 2x the lookups in some pathologically bad cases, do you think will come up rarely, and if so is this method `classTable.IndexOf` linear in complexity or something worse?

File: src/Engine/ProtoCore/Lang/Type.cs (hunk 1 above)
Comment (aparajit-pratap) [reply src/Engine/ProtoCore/Lang/Type.cs:127]: I was thinking that someone could manually type `long`, but that's highly unlikely.

Comment (aparajit-pratap) [reply src/Engine/ProtoCore/Lang/Type.cs:127]: I can remove it.

File: src/Engine/ProtoCore/Lang/Type.cs (hunk 2 above)
Comment (aparajit-pratap) [reply src/Engine/ProtoCore/Lang/Type.cs:330]: I think right now, it's only going to take place for a node like `RemoveIfNot `where there is the possibility of checking for a C# type.

**Comment by mjkkirschner** [comment]
LGTM

**Pull request merged**